src/circuit/num.rs

```diff
@@ -729,10 +729,12 @@ impl<E: Engine> AllocatedNum<E> {
             |lc| lc + a.get_variable() - b.get_variable()
         );
 
-        /// Let 'bits' - 255 bits representation of delta
+        /// Let 'bits' - E::Fr::NUM_BITS bits representation of delta
         let bits = delta.into_bits_le(&mut cs)?;
 
-        match bits[254].get_variable() {
+        assert_eq!(bits.len() as u32, E::Fr::NUM_BITS);
+
+        match bits[(E::Fr::NUM_BITS as usize) - 1].get_variable() {
```

Comment:
How does this Not crash? But decomposition should give a vector of NUM_BITS elements which is zero enumerates, so you actually try to seek pass the end